app/workers/generate_test_applications.rb

```diff
@@ -1,50 +1,83 @@
-    create states: [:withdrawn]
-    create states: [:awaiting_provider_decision], apply_again: true
+    create recruitment_cycle_year: 2020, states: %i[rejected rejected]
+    create recruitment_cycle_year: 2020, states: %i[offer_withdrawn]
+    create recruitment_cycle_year: 2020, states: %i[offer_deferred]
+    create recruitment_cycle_year: 2020, states: %i[offer_deferred]
+    create recruitment_cycle_year: 2020, states: %i[declined]
+    create recruitment_cycle_year: 2020, states: %i[accepted]
+    create recruitment_cycle_year: 2020, states: %i[recruited]
+    create recruitment_cycle_year: 2020, states: %i[conditions_not_met]
+    create recruitment_cycle_year: 2020, states: %i[withdrawn]
+    create recruitment_cycle_year: 2020, states: %i[application_not_sent]
+
+    create recruitment_cycle_year: 2021, states: %i[unsubmitted]
+    create recruitment_cycle_year: 2021, states: %i[unsubmitted], course_full: true
+    create recruitment_cycle_year: 2021, states: %i[awaiting_provider_decision awaiting_provider_decision awaiting_provider_decision]
+    create recruitment_cycle_year: 2021, states: %i[awaiting_provider_decision awaiting_provider_decision awaiting_provider_decision]
+    create recruitment_cycle_year: 2021, states: %i[awaiting_provider_decision awaiting_provider_decision awaiting_provider_decision]
+    create recruitment_cycle_year: 2021, states: %i[awaiting_provider_decision], apply_again: true
+    create recruitment_cycle_year: 2021, states: %i[offer offer]
+    create recruitment_cycle_year: 2021, states: %i[offer rejected]
+    create recruitment_cycle_year: 2021, states: %i[rejected rejected]
+    create recruitment_cycle_year: 2021, states: %i[offer_withdrawn]
+    create recruitment_cycle_year: 2021, states: %i[offer_deferred]
+    create recruitment_cycle_year: 2021, states: %i[offer_deferred]
+    create recruitment_cycle_year: 2021, states: %i[declined]
+    create recruitment_cycle_year: 2021, states: %i[accepted]
+    create recruitment_cycle_year: 2021, states: %i[accepted_no_conditions]
+    create recruitment_cycle_year: 2021, states: %i[recruited]
+    create recruitment_cycle_year: 2021, states: %i[conditions_not_met]
+    create recruitment_cycle_year: 2021, states: %i[withdrawn]
+
+    without_slack_message_sending do
+      RejectApplicationsByDefault.new.call
+    end
   end
 
-  def create(states:, apply_again: false, course_full: false)
-    @test_applications.create_application(
-      courses_to_apply_to: @courses_to_apply_to,
+private
+
+  def create(recruitment_cycle_year: nil, states:, apply_again: false, course_full: false)
+    TestApplications.new.create_application(
       states: states,
+      recruitment_cycle_year: recruitment_cycle_year,
+      courses_to_apply_to: recruitment_cycle_year == 2020 ? courses_to_apply_to_in_2020 : courses_to_apply_to_in_2021,
       apply_again: apply_again,
       course_full: course_full,
     )
   end
+
+  def courses_to_apply_to_in_2020
```

Comment:
From what I've seen, 2020 courses are still `exposed_in_find`, so this and the following method could probably be merged into a common `def courses_to_apply_to_in(recruitment_cycle_year)`.

Reply:
See also https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/lib/tasks/local_dev.rake#L48